### DISCUSSION TITLE
Source for Deprecation Information in Mongoid

### DISCUSSION THREAD

**Discussion by jeffsaremi** [mongodb/mongoid#5334] [Q&A]
I am looking for a place where the deprecated api's are listed. I have looked at the release notes here but I don't see any mentions wrt some api changes. (https://github.com/mongodb/mongoid/releases)
Example: 
it looks like version >= 7.0 has removed, replaced, renamed the Relations module and its sub-modules to Association
However I'm unable to find any announcement or documentation of this significant change.

**Comment by p-mongo** [comment]
Hi @jeffsaremi ,

Deprecations are generally listed in release notes. For example, https://www.mongodb.com/docs/mongoid/master/release-notes/mongoid-7.4/ has a few, and 7.5 will have a bunch more.

Older versions such as 7.0 haven't had the release notes written for them the way we produce them today. If you have specific questions feel free to ask here.

Note that changes that are not in Mongoid's public API generally aren't advertised in release notes. Renaming of internal Mongoid modules is likely to fall in this category.

**Comment by jeffsaremi** [comment]
What you're saying doesn't make sense at all!
What I understood was this:

The breaking changes to `public` interface of mongoid `release`d to the `public` do not need to announced.

Show me one other organization who writes and distributes code like this.

**Reply by alexbevi** [reply]
Hi @jeffsaremi,

We continue to work on improving our documentation, including surfacing changes to public APIs where possible. To ensure we're not overlooking something important we'd like to better understand the specific circumstances leading to this discussion item being raised.

Do you have an example from your application you can share that worked prior to Mongoid 7.0 that broke on upgrade? Were you using any specific API that we changed that caused your team issues?

Mongoid 7.0 was released in 2018 and much has changed since then, however we strive to continue to improve our products and processes and feedback such as this is extremely valuable.

Any additional information you can share would be appreciated.

**Reply by jeffsaremi** [reply]
Thank you!
